nyxx/lib/src/internal/_HttpEndpoints.dart

```diff
@@ -1162,6 +1168,52 @@ class _HttpEndpoints implements IHttpEndpoints {
           "/channels/$channelId/messages/$messageId/crosspost",
           method: "POST"));
 
+  @override
+  Future<ThreadPreviewChannel> createThreadWithMessage(
+      Snowflake channelId, Snowflake messageId, ThreadBuilder builder) async {
+    final response = await _httpClient._execute(BasicRequest._new(
+        "/channels/$channelId/messages/$messageId/threads",
+        method: "POST", body: builder._build(),),);
+
+    if (response is HttpResponseSuccess) {
+      return ThreadPreviewChannel._new(_client, response.jsonBody as Map<String, dynamic>);
+    }
+
+    return Future.error(response);
+  }
+
+
+  @override
+  Future<ThreadPreviewChannel> createThread(
+      Snowflake channelId, ThreadBuilder builder) async {
+    final response = await _httpClient._execute(BasicRequest._new(
+      "/channels/$channelId/threads",
+      method: "POST", body: builder._build(),),);
+
+    if (response is HttpResponseSuccess) {
+      return ThreadPreviewChannel._new(_client, response.jsonBody as Map<String, dynamic>);
+    }
+
+    return Future.error(response);
+  }
+
+  @override
+  Future<List<_MemberCacheable>> getThreadMembers(Snowflake channelId, Snowflake guildId) async {
+    final response = await _httpClient._execute(BasicRequest._new("/channels/$channelId/thread-members", method: "GET",),);
+
+    if (response is HttpResponseSuccess) {
+      final body = response.jsonBody as List<dynamic>;
+      final guild = new _GuildCacheable(_client, guildId);
+      List<_MemberCacheable> members = [];
+      for(final id in body) {
+        members.add(new _MemberCacheable(_client, Snowflake(id), guild));
+      }
+      return members;
```

Comment:
this could be implemented using spread collections. I think that would be more readable and "darty"

Reply:
done